### ISSUE TITLE
Auto Build Specific Version of OpenCV

### ISSUE THREAD

**Issue by LeeGDavis** [justadudewhohacks/opencv4nodejs#399]
Is there a way to specify a specific version of opencv, namely 3.3.1, as apposed to the latest when using the autobuild?

**Comment by LeeGDavis** [comment]
There are breaking changes to how it is processing images (could be our usage of it - doesn't matter) and upgrading and fixing to consume opencv 3.4.x is not on the table at this point.  I was hoping for a env var like OPENCV_BUILD_VERSION=3.3.1.

**Comment by justadudewhohacks** [comment]
If your code is depending on a specific OpenCV version, then you shouldn't use the auto build in my opinion. It is fairly easy to build OpenCV 3.3.1 on your own.

**Comment by LeeGDavis** [comment]
I agree, in fact I'm on windows and it is trivial (especially with prebuild binaries - however manually building just works too).  I however have colleagues that are struggling on OSX with various version of xcode yielding different results.   We don't know exactly what the issue is, however that is a build issue with OpenCV.   I could open a PR on your build repo to allow a ENV variable to build a specific version if you are interested given time, but do please let me know if you are not interesting.  I'm also sure it is not that simple, as different build requirements likely exist between versions....

**Comment by justadudewhohacks** [comment]
I mean, it doesn't hurt to allow for such an env and have it default to 3.4.1.

**Comment by LeeGDavis** [comment]
@justadudewhohacks it looks like the crux of the issue we were running into on osx was https://github.com/opencv/opencv/pull/10011 and we could work around some other issues different versions of xcode introduced.  Auto building still yields the same error and has since been resolved on newer versions of opencv, which we have yet to implement. Selfishly there is not a lot of value for us at this point to add support to fix this in the autobuild, given the solution is the same for both autobuild or manually building.  However if you see a PR on this in the future you'll no we changed out tune.  :D Thanks again!
